common/archive-impl/impl2/src/java/org/sakaiproject/archive/impl/SiteMerger.java

```diff
@@ -309,7 +320,7 @@ else if (element.getTagName().equals(UserDirectoryService.APPLICATION_ID))
 						        if (checkSakaiService(filterSakaiService, filteredSakaiService, serviceName)) {
 						            // checks passed so now we attempt to do the merge
 		                            log.debug("Merging archive data for {} ({}) to site {}", serviceName, fileName, siteId);
-		                            msg = service.merge(siteId, element, fileName, fromSite, attachmentNames, new HashMap() /* empty userIdTran map */, usersListAllowImport);
+		                            msg = service.mergeWithCreatorId(siteId, element, fileName, fromSite, attachmentNames, new HashMap() /* empty userIdTran map */, usersListAllowImport, creatorId);
```

Comment:
Use the old merge if the creatorId is not used here: https://github.com/sakaiproject/sakai/pull/13073/files#diff-6d94998926dbeaaae9d6df97c7f3b42421b65a00c5a2eddffe02f902d0b1633fR128

Reply:
@jesusmmp  Thanks for the review.  The idea is to always call the new method signature in SiteMerger and for implementations that only have the old method signature the `default` code in the api for `mergeWithCreatorId` just calls the old signature without the `creatorId`.  Many of the `merge()` implementations do not need creatorId - but as a weird example the Polls `merge` needs `creatorId`.  This way as we go through and improve all the `archive()` and `merge()` cod - each imlp can decide if it wants `creatorId`.  his saves changing the 10 or so impls that don't need `creatorId` - I am trying to make this as clean a cut as possible to maximize the likelyhood of things being back ported.  

Did I answer your question?  Let me know.

Reply:
@jesusmmp I can't put an if-then-else here.   There is no way that this line of code can know which of the two methods the `merge()` code can be called - it needs to call the more capable method signature and depending on which one is overridden in the service implementation the right one gets called.  @adrianfish mentioned that I should have two method signatures with the same name and differing numbers of parameters - but the compiler complained when I tried that approach (which was my first approach).  I think it is because of the way we register these things and then get them back - but it surprised me.  I will check again.

Reply:
Totally agree with @adrianfish, same name but differing numbers of parameters. It could be great if we can do it on this way. Thanks.